photutils/psf/epsf.py

```diff
@@ -188,34 +188,28 @@ def _fit_star(self, epsf, star, fitter, fitter_kwargs,
             x0 = 0
             y0 = 0
 
-        x_oversamp = star.pixel_scale[0] / epsf.pixel_scale[0]
-        y_oversamp = star.pixel_scale[1] / epsf.pixel_scale[1]
-        scaled_data = data / (x_oversamp * y_oversamp)
+        scaled_data = data / epsf._oversampling**2
 
         # define positions in the ePSF oversampled grid
         yy, xx = np.indices(data.shape, dtype=np.float)
-        xx = (xx - (star.cutout_center[0] - x0)) * x_oversamp
-        yy = (yy - (star.cutout_center[1] - y0)) * y_oversamp
+        xx = (xx - (star.cutout_center[0] - x0)) * epsf._oversampling
+        yy = (yy - (star.cutout_center[1] - y0)) * epsf._oversampling
 
         # define the initial guesses for fitted flux and shifts
         epsf.flux = star.flux
         epsf.x_0 = 0.0
         epsf.y_0 = 0.0
 
-        # The oversampling factor is used in the FittableImageModel
-        # evaluate method (which is use when fitting).  We do not want
-        # to use oversampling here because it has been set by the ratio
-        # of the ePSF and EPSFStar pixel scales.  This allows for
-        # oversampling factors that differ between stars and also for
-        # the factor to be different along the x and y axes.
-        epsf._oversampling = 1.
+        # create copy to avoid overwriting original oversampling factor
+        _epsf = epsf.copy()
```

Comment:
The copy is necessary so that, without removing the non-zero oversampling factor from ``epsf``, we can change the oversampling factor to 1 for the ``fitter`` call below (new line 208). Without the copy, the oversampling factor is lost and subsequent calls to ``_fit_star`` (the loop in ``__call__``) would have ``epsf._oversampling = 1`` for their calculations of ``xx``, ``yy``, ``x_center``, and ``y_center`` and incorrectly determine their values.

This was avoided with the ``pixel_scale`` by calculating ``x_oversamp`` directly from the ratio of the pixel scales of ``star`` and ``epsf``, so it did not matter previously that ``epsf._oversampling`` was effectively reset to one each time, but the removal of ``pixel_scale`` revealed this side effect.